doc/Language-Reference.adoc

```diff
@@ -0,0 +1,1625 @@
+message_type_name ::= xref:syntax-qualified_name[qualified_name]
+[[syntax-qualified_name]]
+qualified_name ::= xref:syntax-name[name] { **.** xref:syntax-name[name] }
+[[syntax-condition]]
+condition ::= xref:syntax-boolean_expression[boolean_expression]
+----
+
+*Static Semantics*
+
+A type refinement describes under which condition a specific protocol message can be expected inside of a payload field. Only components of type `Opaque` can be refined. Types defined in other packages are referenced by a qualified name in the form package_name.message_type_name. The condition can refer to components of the refined type. To indicate that a refined component is empty (i.e. does not exit) under a certain condition, a null message can be used as message type.
+
+*Example*
+
+[source,ada,rflx,basic_declaration]
+----
+for Ethernet::Frame use (Payload => IPv4::Packet)
+   if Ether_Type = Ethernet::IPV4
+----
+
+=== Type Derivation
+
+A type derivation allows to create a new message type based on an existing message type.
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-type_derivation]]
+type_derivation ::= *type* xref:syntax-name[name] *is new* xref:syntax-base_type_name[base_type_name]
+[[syntax-base_type_name]]
+base_type_name ::= xref:syntax-qualified_name[qualified_name]
+----
+
+*Static Semantics*
+
+A derived message type derives its specification from a base type. Type refinements of a base message type are not inherited by the derived message type.
+
+*Example*
+
+[source,ada,rflx,basic_declaration]
+----
+type Specific_Extension is new Extension
+----
+
+=== Sequence Type
+
+A sequence type represents a list of similar elements.
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-sequence_type]]
+sequence_type ::= *type* xref:syntax-name[name] *is sequence of* element_type_xref:syntax-name[name]
+----
+
+*Static Semantics*
+
+A sequence consists of a number of elements with similar type. Scalar types as well as message types can be used as element type. When a sequence is used in a message type, its bit length has to be specified by a size aspect.
+
+*Example*
+
+[source,ada,rflx,basic_declaration]
+----
+type Options is sequence of Option
+----
+
+=== Protocol Sessions [§S]
+
+A session defines the dynamic behavior of a protocol using a finite state machine. The external interface of a session is defined by parameters. The initial and final state is defined by aspects. The declaration part enables the declaration of session global variables. The main part of a session definition are the state definitions.
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-session]]
+session ::=
+   *generic*
+    { xref:syntax-session_parameter[session_parameter] }
+   *session* xref:syntax-name[name] *with*
+      *Initial =>* state_xref:syntax-name[name],
+      *Final =>* state_xref:syntax-name[name]
+   *is*
+    { xref:syntax-session_declaration[session_declaration] }
+   *begin*
+    { xref:syntax-state[state] }
```

Comment:
I have changed the grammar in a1fbddd94dc06e0e846bc4fc92c8b8f5129df9ef, so that at least one state needs to be defined.

A session with just one state is also accepted as a valid model:

```ada
   generic
   session Session with
      Initial => Error,
      Final => Error
   is
   begin
      state Error is null state;
   end Session;
```

I don't think that this is an issue.

Reply:
You're right. While hardly useful, there is no reason to disallow a single state.